### DISCUSSION TITLE
Textadept (semi)knows how to `go build`, but where is that info stored?

### DISCUSSION THREAD

**Discussion by metaleap** [orbitalquark/textadept#144] [Q&A] [answered]
As the Manual says:

> Textadept knows most of the commands that compile and/or run code in source files.

For Go this knowing is currently sub-optimal and I'd make a PR for that. Where to look? Didn't seem to find it in the built-in Go lexer nor in run.lua.

I can also just summarize what I would see as needing tweaking:

- currently it seems to do `go build $file.go`, in most Go packages should better be `go build -o $filedirpath/$filedirname $filedirpath/*.go`
- currently it seems to do `go run $file.go`, more successful for most cases would be `go run $filedir/*.go`

**Comment by rgieseke** [comment, marked as answer]
Here i think: https://github.com/orbitalquark/textadept/blob/ce0da94145eb1faf155e586abddbae70f4f3bd46/modules/textadept/run.lua#L219

`go='go build "%f"'`

**Reply by metaleap** [reply]
Nice thanks!

**Comment by metaleap** [comment]
Aww no shell expansions for the returned shell command in a `compile_commands` function... somehow would need to bring in the `$(basename %d)` or some such

**Reply by rgieseke** [reply]
The compile command can be a function, so you can have the full power of Lua/Textadept.

https://orbitalquark.github.io/textadept/api.html#textadept.run.compile_commands

**Reply by metaleap** [reply]
I know and that's cool but just would have been faster to `basename %d` rather than look up all the file-system-path-related apis in yet-another-new-to-me-scripting-lang  =)  well there's always the option to `sh -c my command line` to bring expansions back in, wastefully mad as it seems :grin:
